### PR TITLE
made solution crawler to prefer project with errors as next project t…

### PR DESCRIPTION
**Pull request by heejaechang**
…o process.

this should let us remove errors faster than introducing ones.

### CODE REVIEW

**Comment by heejaechang** [comment]
@mavasani @srivatsn can you take a look?

File: src/Features/Core/Portable/Diagnostics/EngineV1/DiagnosticIncrementalAnalyzer.cs [hunk 1]
```diff
@@ -511,6 +510,43 @@ public override void RemoveProject(ProjectId projectId)
...
+            {
+                for (var stateType = 0; stateType < s_stateTypeCount; stateType++)
+                {
+                    var state = stateSet.GetState((StateType)stateType);
+                    if (state.Count == 0)
+                    {
+                        continue;
+                    }
+
+                    if (state.GetDataCount(projectId) > 0)
```
Comment (mavasani) [inline src/Features/Core/Portable/Diagnostics/EngineV1/DiagnosticIncrementalAnalyzer.cs]: Should we ignore hidden diagnostics?

File: src/Features/Core/Portable/Diagnostics/EngineV2/DiagnosticIncrementalAnalyzer.cs [hunk 2]
```diff
@@ -224,5 +224,11 @@ private void RaiseEvents(Project project, ImmutableArray<DiagnosticData> diagnos
                         ValueTuple.Create(this, kv.Key), workspace, solution, project.Id, kv.Key, kv.ToImmutableArrayOrEmpty()));
             }
         }
+
+        public override bool ContainsDiagnostics(Workspace workspace, ProjectId projectId)
+        {
+            // for now, it always return false;
```
Comment (mavasani) [inline src/Features/Core/Portable/Diagnostics/EngineV2/DiagnosticIncrementalAnalyzer.cs]: At some point we need to either delete EngineV2 OR reuse it for your out-of-proc work.

File: src/Features/Core/Portable/Diagnostics/EngineV1/DiagnosticIncrementalAnalyzer.cs (hunk 1 above)
Comment (heejaechang) [inline src/Features/Core/Portable/Diagnostics/EngineV1/DiagnosticIncrementalAnalyzer.cs]: I thought about that, but for this one, what it is doing already seems expensive. so no, but in V2, will do.

File: src/Features/Core/Portable/Diagnostics/EngineV2/DiagnosticIncrementalAnalyzer.cs (hunk 2 above)
Comment (heejaechang) [inline src/Features/Core/Portable/Diagnostics/EngineV2/DiagnosticIncrementalAnalyzer.cs]: I am going to use v2 engine for oop work.

**Comment by mavasani** [comment]
LGTM

**Comment by heejaechang** [comment]
@dotnet-bot retest prtest/mac/dbg/unit32 please

**Comment by heejaechang** [comment]
@dotnet/roslyn-infrastructure mac timed out again.

**Comment by heejaechang** [comment]
@dotnet/roslyn-infrastructure it looks like test doesn't progress?

**Comment by heejaechang** [comment]
retest prtest/win/dbg/unit32 please

**Comment by heejaechang** [comment]
retest prtest/win/dbg/unit64  please

**Comment by heejaechang** [comment]
infra team told me to retest.

**Comment by davkean** [comment]
I don't think it's a coincidence that *both* x86 and x64 tests ran into deadlocks. These are probably caused your by changes.

**Pull request merged**